src/uu/cp/src/cp.rs

```diff
@@ -710,43 +738,42 @@ impl Options {
         };
 
         // Parse attributes to preserve
-        let mut preserve_attributes: Vec<Attribute> = if matches.contains_id(options::PRESERVE) {
+        let attributes: Attributes = if matches.contains_id(options::PRESERVE) {
             match matches.get_many::<String>(options::PRESERVE) {
-                None => DEFAULT_ATTRIBUTES.to_vec(),
+                None => Attributes::default(),
                 Some(attribute_strs) => {
-                    let mut attributes = Vec::new();
+                    let mut attributes: Attributes = Attributes::none();
+                    let mut attributes_empty = true;
                     for attribute_str in attribute_strs {
+                        attributes_empty = false;
                         if attribute_str == "all" {
-                            attributes = add_all_attributes();
+                            attributes = Attributes::all();
                             break;
                         } else {
-                            attributes.push(Attribute::from_str(attribute_str)?);
+                            attributes.try_set_from_string(attribute_str, Attributes::default())?;
```

Comment:
It should still error because it's basically disabled on a non-SELinux machine (because of the attribute). Which doesn't make it easier, hm. Could it be an issue if you try using a binary built on a SELinux machine on a non-SELinux machine?